File: HelpSoftware/Help/Help.EF/Entitys/UserWatchArticle.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Help.EF
{
  public class UserWatchArticle : IHelpEntity
  {
    [Key]
    public virtual int ID_UserWatchArticle { get; set; }

    public virtual User WatchedUser { get; set; }

    public virtual Article WatchedArticle { get; set; }

    public virtual DateTime WatchedTime { get; set; }

    [Timestamp]
    public byte[ ] TimeStamp { get; set; }

    public string GetEntityName()
    {
      return "UserWatchArticle";
    }

    public int GetID()
    {
      return ID_UserWatchArticle;
    }

    public bool GetWriteHistoryEntry()
    {
      return false;
    }
  }
}

File: HelpSoftware/Help/Help.Web/Web/src/app/EF/UserWatchArticle.ts
import { User, Article } from "./EF";

export class UserWatchArticle {
  ID_UserWatchArticle: number;
  WatchedUser: User;
  WatchedArticle: Article;
  WatchedTime: Date;

  deserialize(input): UserWatchArticle {
    this.ID_UserWatchArticle = input.ID_UserWatchArticle;
    this.WatchedUser = new User().deserialize(input.WatchedUser);
    this.WatchedArticle = new Article().deserialize(input.WatchedArticle);
    this.WatchedTime = input.WatchedTime;
    return this;
  }
}
